node/src/components/storage.rs

```diff
@@ -284,15 +286,24 @@ impl Storage {
         info!("reindexing block store");
         let mut block_height_index = BTreeMap::new();
         let mut switch_block_era_id_index = BTreeMap::new();
-        let block_txn = env.begin_ro_txn()?;
-        let mut cursor = block_txn.open_ro_cursor(block_header_db)?;
+        let mut block_txn = env.begin_rw_txn()?;
```

Comment:
I think we currently drop read-only transactions rather than committing them, so this should be committed  where we currently drop it.

Reply:
You mean we need to commit it explicitly in order for the deletion to take effect?

Reply:
Yes, I'd expect so.

Reply:
You're right.

Reply:
That makes sense :+1: